src/read/wasm.rs

```diff
@@ -56,23 +71,227 @@ impl<'data> WasmFile<'data> {
 
         let mut file = WasmFile::default();
 
+        let mut main_file_symbol = Some(Symbol {
+            name: None,
+            address: 0,
+            size: 0,
+            kind: SymbolKind::File,
+            section: SymbolSection::None,
+            weak: false,
+            scope: SymbolScope::Compilation,
+            flags: SymbolFlags::None,
+        });
+
+        let mut imported_funcs_count = 0;
+        let mut local_func_kinds = Vec::new();
+        let mut entry_func_id = None;
+
         for section in module {
             let section = section.ok().read_error("Invalid Wasm section header")?;
```

Comment:
Now that you've implemented `ReadError` (also for the call to `ModuleReader::new`):

```suggestion
            let section = section.read_error("Invalid Wasm section header")?;
```